### DISCUSSION TITLE
[Question] IInputContext + IWindow + GL

### DISCUSSION THREAD

**Discussion by tilkinsc** [dotnet/Silk.NET#1225] [Q&A] [answered]
Is it legal to call IWindow.CreateInput() and set events directly after Window.Create(WindowOptions) instead of in the Load event function like the examples in this repository do? Is it also legal to create the GL context after IWindow.Run() instead of in the Load event?

**Comment by Perksey** [comment, marked as answer]
It is illegal to create an input context for an uninitialized window due to internal state not being guaranteed. It needs to be after `Initialize`/`Run`

Likewise, it is illegal to call `window.CreateOpenGL`/`GL.GetApi` with an uninitialized window due to the state of IGLContextSource.Context not being guaranteed. However, you can call it any time after `Initialize`/`Run`.

Hope this helps.

**Reply by tilkinsc** [reply]
Thanks Perksey it answered my question!
